source/rtcp_api.c

```diff
@@ -789,17 +782,24 @@ RtcpResult_t Rtcp_DeserializePacket( RtcpContext_t * pCtx,
 
     if( result == RTCP_RESULT_OK )
     {
-        pRtcpPacket->header.padding = ( firstWord & RTCP_HEADER_PADDING_BITMASK ) >>
-                                      RTCP_HEADER_PADDING_LOCATION;
-
-         /* RC field is FMT in some Application Feedback and Transport Feedback
-          * messages. */
+        /* RC field is FMT in some Application Feedback and Transport Feedback
+         * messages. */
         fmt = ( firstWord & RTCP_HEADER_RC_BITMASK ) >>
               RTCP_HEADER_RC_LOCATION;
         packetType = ( firstWord & RTCP_HEADER_PACKET_TYPE_BITMASK ) >>
                      RTCP_HEADER_PACKET_TYPE_LOCATION;
 
-        pRtcpPacket->header.packetType = GetRtcpPacketType( packetType, fmt );
+        rtcpPacketType = GetRtcpPacketType( packetType, fmt );
+
+        if( rtcpPacketType == RTCP_PACKET_UNKNOWN )
+        {
+            result = RTCP_RESULT_MALFORMED_PACKET;
+        }
+
+        pRtcpPacket->header.padding = ( firstWord & RTCP_HEADER_PADDING_BITMASK ) >>
+                                      RTCP_HEADER_PADDING_LOCATION;
+
```

Comment:
Can you please help me understand if it is a malformed packet why are we continuing to parse it? like reception count, payload and all the other fields? 
I think we should exit without parsing as we are not sure of the contents of the packet.

Reply:
Good catch. Addressed in the latest commit.